server/src/utils/vm/environment.ts

```diff
@@ -28,9 +31,24 @@ export class EnvironmentProvider {
   }
 
   private buildEnvironmentFrom(files: Parameters<typeof buildEnvironment>[0], baseEnvironment?: Environment): Environment {
-    this.buildProgressReporter.begin()
-    const environment = buildEnvironment(files, baseEnvironment)
-    this.buildProgressReporter.end()
-    return environment
+    try {
+      this.buildProgressReporter.begin()
+      const timeMeasurer = new TimeMeasurer()
+      const environment = buildEnvironment(files, baseEnvironment)
+      timeMeasurer.addTime('Building environment')
+      inferTypes(environment)
+      timeMeasurer.addTime('Infering types')
```

Comment:
```suggestion
      timeMeasurer.addTime('Inferring types')
```

Una boludez, pero es más fuerte que yo